testomania/src/main/java/com/earth/testomania/common/networking/NetworkConnectivityObserver.kt

```diff
@@ -0,0 +1,56 @@
+package com.earth.testomania.common.networking
+
+import android.content.Context
+import android.net.*
+import dagger.hilt.android.qualifiers.ApplicationContext
+import kotlinx.coroutines.channels.awaitClose
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.callbackFlow
+import kotlinx.coroutines.flow.distinctUntilChanged
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/**
+ * Created by Vazhapp on 23.10.2022
+ */
+@Singleton
+class NetworkConnectivityObserver @Inject constructor(
+    @ApplicationContext context: Context,
+) : ConnectivityObserver {
+
+    private var connectivityManager: ConnectivityManager =
```

Comment:
Nit: ConnectivityManager type is not needed since you are casting to ConnectivityManager anyway